subprojects/core/src/main/java/org/gradle/internal/model/LoadingCache.java

```diff
@@ -16,22 +16,24 @@
 
 package org.gradle.internal.model;
 
-import java.util.Map;
-import java.util.function.Function;
-
 /**
- * A cache backed by calculated values. This map is thread-safe.
+ * A cache that loads values on demand as they are requested.
  */
-public interface CalculatedValueCache<K, V> {
+public interface LoadingCache<K, V> {
```

Comment:
Is it worth giving this a more unique name? We already have Guava on the classpath in some projects, and I can see a mixup where one of those is used instead of our `LoadingCache`. Or perhaps we should just add a checkstyle rule to ban using that cache instead.

Reply:
I'm open to a more unique name.

Ideally we'd use this everywhere instead of Guava's `LoadingCache`, though I don't want to propagate this everywhere atm.

The end goal here is a centralized cache factory, where we can track all the caches used in the entire build, therefore allowing us to debug caches by turning on statistics.

Reply:
Renamed to `InMemoryLoadingCache`